src/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -418,10 +428,15 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
             Map<Id, npe03__Recurring_Donation__c> oldRdsById = new Map<Id, npe03__Recurring_Donation__c>(oldRds);
 
             for (npe03__Recurring_Donation__c rd : rds) {
-                if (evalService.hasKeyFieldChanged(rd, oldRdsById.get(rd.Id))) {
+                npe03__Recurring_Donation__c oldRd = oldRdsById.get(rd.Id);
+
+                if (evalService.hasKeyFieldChanged(rd, oldRd)) {
                     rdIds.add(rd.Id);
                 }
-                if (rd.npe03__Next_Payment_Date__c != oldRdsById.get(rd.Id).npe03__Next_Payment_Date__c) {
+
+                if (rd.npe03__Next_Payment_Date__c != oldRd.npe03__Next_Payment_Date__c 
```

Comment:
Adding the check for fields changing the schedule fixed the issue tested in [this test](https://github.com/SalesforceFoundation/NPSP/blob/5a7e6004051e595592f39c6a3d3d3fe542445535/src/classes/RD2_OpportunityService_TEST.cls#L619). If the Effective Date (`StartDate__c`) is not set, then test pass w/o the fix, however setting the Effective Date uncovers an error  that is fixed by retrieving Opportunities to calculate the NDD and then as a fail safe, check fields changing the schedule to see if Opportunities need to be evaluated.